src/Hooks/CommandStartingListener.php

```diff
@@ -62,26 +65,26 @@ private function registerJobHooks(): void
         $this->nightwatch->configureForJobs();
 
         /**
+         * @see \Laravel\Nightwatch\Core::ingest()
          * @see \Laravel\Nightwatch\State\CommandState::reset()
-         */
-        $this->events->listen(JobPopping::class, (new JobPoppingListener($this->nightwatch))(...));
-
-        /**
          * @see \Laravel\Nightwatch\State\CommandState::$timestamp
          * @see \Laravel\Nightwatch\State\CommandState::$id
          */
-        $this->events->listen(JobProcessing::class, (new JobProcessingListener($this->nightwatch))(...));
-
-        /**
-         * @see \Laravel\Nightwatch\Records\Exception
-         */
-        $this->events->listen(JobExceptionOccurred::class, (new JobExceptionOccurredListener($this->nightwatch))(...));
+        $this->events->listen([
+            Looping::class,
+            JobPopping::class,
+            JobProcessing::class,
+            WorkerStopping::class,
+        ], (new WorkerEventListener($this->nightwatch))(...));
```

Comment:
I've consolidated the event listeners for the `JobPopping` and `JobProcessing` events. It also handles the `Looping` and `WorkerStopping` events to ingest the data.